website/server/models/group.js

```diff
@@ -750,15 +750,18 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
     $inc: {memberCount: -1},
   };
 
-  let challenges = await Challenge.find({
-    _id: {$in: user.challenges},
-    group: group._id,
-  });
+  // only remove user from challenges if it's set to remove-all
+  if (keep === 'remove-all') {
```

Comment:
I've updated the documentation here. I left the "defaults to keep-all" section because that's still true: https://github.com/HabitRPG/habitrpg/pull/7781/files#diff-b6f239870392ed0ec2a2b73cbebaa699R747